docs/rules/no-disallowed-lwc-imports.md

```diff
@@ -0,0 +1,41 @@
+## Prevent importing restricted APIs from the "lwc" package (no-disallowed-lwc-imports)
+
+Restricts importing disallowed modules from the `lwc` package. It's recommended to only use the "safe" APIs allowed by this rule.
+
+This rule also disallows default imports as well as bare imports and exports – only importing explicitly named modules is allowed.
+
+### Rule details
+
+Examples of **incorrect** code:
+
+```js
+import { SecretApiYouShouldNotUse } from 'lwc';
+```
+
+```js
+import * as lwc from 'lwc';
+```
+
+```js
+import lwc from 'lwc';
+```
+
+```js
+import 'lwc';
+```
+
+```js
+export * from 'lwc';
+```
+
+Examples of **correct** code:
+
+```js
+import { LightningElement } from 'lwc';
+```
+
+```js
+import { LightningElement, wire, api } from 'lwc';
+```
+
+If you disable this rule, then you may import unstable or otherwise undesirable APIs from `lwc`.
```

Comment:
Shall we make this rule configurable to allow importing unstable API? Maybe it's something we can include in the future if needed.

```js
{
  rules: {
    'no-disallowed-lwc-imports': ["error", { allowUnstable: true }]
  }
}
```

or: 

```js
{
  rules: {
    'no-disallowed-lwc-imports': ["error", { allow: ['swapComponent'] }]
  }
}
```

Reply:
This seems unnecessary to me. People who want to use unstable imports can disable the rule entirely.

The value-add of the other things that the rule checks is low IMO. Everything else that it checks for (e.g. `import nonexistentDefault from 'lwc'`) will also throw an error either by Rollup or at runtime.

Reply:
Sounds good to me.